src/main/java/cloud/eppo/SdkKey.java

```diff
@@ -0,0 +1,80 @@
+package cloud.eppo;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Wrapper for an SDK key; built from the SDK Key token string, this class extracts encoded fields,
+ * such as the customer-specific service gateway subdomain
+ */
+public class SdkKey {
+  private final String sdkKey;
+  private final Map<String, String> decodedParams;
+
+  public SdkKey(String sdkKey) {
+    this.sdkKey = sdkKey;
+    this.decodedParams = decodeToken(sdkKey);
+  }
+
+  private Map<String, String> decodeToken(String token) {
+    try {
+      String[] parts = token.split("\\.");
+      if (parts.length < 2) {
+        return Collections.emptyMap();
+      }
+
+      String payload = parts[1];
+      if (payload == null) {
+        return Collections.emptyMap();
+      }
+
+      String decodedString = Utils.base64Decode(payload);
+      final Map<String, String> query_pairs = new LinkedHashMap<>();
```

Comment:
Any reason for a `LinkedHashMap` vs. a regular `HashMap`? I don't see us iterating over this thing anywhere.

Reply:
not particulary